tests/pytest/oauth/test_client.py

```diff
@@ -16,14 +16,37 @@ def test_instance_no_oauth_client_name():
 
 
 @pytest.mark.django_db
-def test_instance_oauth_client_name():
-    assert not client._OAUTH_CLIENT
-
+def test_instance():
     oauth_client = client.instance()
+    oauth_client2 = client.instance()
 
     assert oauth_client
-    assert client._OAUTH_CLIENT is oauth_client
+    assert oauth_client2
+    assert oauth_client is not oauth_client2
 
-    oauth_client2 = client.instance()
 
-    assert oauth_client is oauth_client2
+@pytest.mark.django_db
+def test_instance_scope():
+    scope1 = "scope1"
+    oauth_client = client.instance(scope1)
+    client_scope = oauth_client.client_kwargs["scope"]
+
+    assert scope1 in client_scope
+    assert client._OPENID_SCOPE in client_scope
+
+    scope2 = "scope2"
+    oauth_client2 = client.instance(scope2)
+    client_scope = oauth_client2.client_kwargs["scope"]
+
+    assert scope1 not in client_scope
+    assert scope2 in client_scope
+    assert client._OPENID_SCOPE in client_scope
```

Comment:
Hmm, not sure I would check internal properties like this. Could we generate the redirect URL and check that?